### PR TITLE
Construction Bugfixes

### PR DESCRIPTION
**Pull request by Acruid**
* Added CanReach property to AfterAttack events which signals if the attack was within reach.
* You cannot construct/deconstruct things out of reach.
* Construction entities now preserve transform rotation.
* No more exceptions about missing grids when constructing world entities.
* Used the proper intermediate sprite for intermediate entities.
* Issue with missing sprite layers on ghost.

Resolves https://github.com/space-wizards/space-station-14/issues/1308.
Resolves https://github.com/space-wizards/space-station-14/issues/1317.

### CODE REVIEW

**Comment by ike709** [comment]
No fix for https://github.com/space-wizards/space-station-14/issues/1296 ?

**Comment by Acruid** [comment]
Yeh, this resolves https://github.com/space-wizards/space-station-14/issues/1296 also.

File: Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs [hunk 1]
```diff
@@ -286,7 +286,8 @@ private void UserInteraction(IEntity player, GridCoordinates coordinates, Entity
                 if (item != null)
                 {
                     // After attack: Check if we clicked on an empty location, if so the only interaction we can do is AfterInteract
-                    InteractAfter(player, item, coordinates);
+                    var distSqrt = (playerTransform.WorldPosition - coordinates.ToMapPos(_mapManager)).LengthSquared;
```
[the remark this answers is not in the document]
Comment (Acruid) [reply Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs:289]: The `collisionMask`, `ignoredEnt`, and `ignoreInsideBlocker` arguments for `InRangeUnobstructed` seem like they are specific to how each event handler deals with the event. If that is the case, it is not possible to put that check here, and each handler needs to independently perform the check.

[the remark this answers is not in the document]
Comment (Zumorica) [reply Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs:289]: I guess that's true. We should put `InRangeUnobstructed` checks in each event handler then. After all, we don't want people constructing/deconstructing things through walls :cold_sweat:

[the remark this answers is not in the document]
Comment (Acruid) [reply Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs:289]: Good idea.

**Pull request merged**